app/ui/src/main/java/com/fsck/k9/activity/MessageList.java

```diff
@@ -1688,31 +1692,32 @@ private void unlockDrawer() {
 
     private void initializeFromLocalSearch(LocalSearch search) {
         this.search = search;
+        singleFolderMode = false;
 
         if (search.searchAllAccounts()) {
-            List<Account> accounts = preferences.getAccounts();
-            singleAccountMode = (accounts.size() == 1);
-            if (singleAccountMode) {
-                account = accounts.get(0);
-            }
+            account = null;
         } else {
             String[] accountUuids = search.getAccountUuids();
-            singleAccountMode = (accountUuids.length == 1);
-            if (singleAccountMode) {
+            if (accountUuids.length == 1) {
```

Comment:
This doesn't set `account` to null in case the condition isn't met.

Reply:
I was already wondering under what conditions this situation occurs and what should happen then.

OK, the latter is answered and more important. Will add an else-branch to favour legibility over elegance.